File: BITEC/lessons/block_2_object_orientation/src/lerneinheiten/L02KlassenUndMethoden/klassen/Pudel.java
package lerneinheiten.L02KlassenUndMethoden.klassen;

public class Pudel extends Hund {
    double fluff;

    public Pudel(String name, Double happiness, Double health, boolean chipped, Integer alter, HundeBesitzer besitzer, double fluff) {
        super(name, happiness, health, chipped, alter, besitzer);
        this.fluff = fluff;
    }

    public void winseln() {
        System.out.println(".... ewww");
        setHealth(getHealth() - 1);
    }

    public double getFluff() {
        return fluff;
    }

    public void setFluff(double fluff) {
        this.fluff = fluff;
    }
}

File: C#/C#.NET/modul_1_grundlagen/L05KlassenMethoden/Exercise1-HundeUndBesitzer/Pudel.cs
namespace Hunde;

public class Pudel : Hund // Ist-Beziehungen
{
    // private Felder
    private double _fluff;

    // Konstruktor
    public Pudel(
        string name, int alter, string geschlecht, 
        double health, bool chipped, double fluff
    ) : base(
            name, alter, geschlecht, health, chipped
        )
    {
        _fluff = fluff;
        SetDarstellung("🐩");
        SetLautBeimBellen("⬆️Ieek⬆️");
    }

    public Pudel(
        string name, int alter, string geschlecht, 
        double health, bool chipped, double fluff, 
        HundeBesitzer besitzer, Hund spielFreund
    ) : this(
            name, alter, geschlecht, health, chipped, fluff
        )
    {
        SetBesitzer(besitzer);
        SetSpielFreund(spielFreund);
    }

    public Pudel(
        string name, int alter, string geschlecht, double health, 
        bool chipped, double fluff, HundeBesitzer besitzer
    ) : this(
            name, alter, geschlecht, health, chipped, fluff
        )
    {
        SetBesitzer(besitzer);
    }

    public Pudel(
        string name, int alter, string geschlecht, 
        double health, bool chipped, double fluff, 
        Hund spielFreund
    ) : this(
            name, alter, geschlecht, health, chipped, fluff
        )
    {
        SetSpielFreund(spielFreund);
    }

    // Methoden
    public void Winseln()
    {
        Console.WriteLine(".winsel.");
        SetHealth(GetHealth() - 1);
    }

    // Get-und-Set-Methoden
    public double GetFluff()
    {
        return _fluff;
    }

    public void SetFluff(double fluff)
    {
        _fluff = fluff;
    }
}